module-4/backend/week_four.md

```diff
@@ -1,19 +1,55 @@
 ## Week Four - Module 4 Recap
 
-Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
+1. What's your favorite project management tool?
 
-Note: When you're done, submit a PR. 
+I like Waffle for its Git integration and its less busy compared to Pivotal Tracker, especially for individual or pair projects. I've recently been introduced to Jira, and would be interested in using it for its Git and CI integration.  
 
-1. What's your favorite project management tool?
 2. Why do we create staging environments?
+
+To test how the software works with the code changes made but in an environment that is the same as the production environment instead of the local dev invironement/s. Ultimately, to prevent introducing bugs and breaking functionality upon deployment.
+
 3. What are the characteristics of a good README (in your opinion)?
+
+It should describe the project and its purpose. It should include instructions for set up and dependencies. Also potentially instructions for how to contribute in the project and pictures/snippets for ease of use. 
+
 4. What's one main improvement you're going to make to your code regarding accessibility issues?
+
+Practice use of appropriate HTML elements
+Make forms accessible by including descriptive tags
+Minimize changing fonts and beware of distracting color themes
+Learn to use ARIA roles and alerts for dynamically updated content
+
 5. What are some basic security concerns to be aware of when building applications?
+
+In Rails, use has_secure_password method with password_digest attribute to authenticate user login against bcrypt password 
+Don't share sensitive information on GitHub (production keys and authentication tokens)
+Don't pass information from URL params and forms into query without sanitizing || AR ORM layer is safe against SQL injection
+Make methods private whenever possible to expose only necessary app functionality
+
 6. Why is continuous integration helpful/important?
+
+CI is an automated tool to help in tracking how changes to the code base may introduce bugs and in doing so, to reduce potential debugging time for code built on top of code that wasn't working as anticipated. Like staging, it tests the code in production ENV. CI can't work without thorough unit and integration testing.
```

Comment:
yup! it runs the entire test suite against a production-like environment to ensure you haven't introduced any larger bugs!